apps/web/utils/actions/settings.ts

```diff
@@ -34,12 +37,86 @@ export const updateAiSettingsAction = actionClientUser
       ctx: { userId },
       parsedInput: { aiProvider, aiModel, aiApiKey },
     }) => {
-      await prisma.user.update({
-        where: { id: userId },
-        data:
-          aiProvider === DEFAULT_PROVIDER
-            ? { aiProvider: null, aiModel: null, aiApiKey: null }
-            : { aiProvider, aiModel, aiApiKey },
-      });
+      try {
+        await prisma.user.update({
+          where: { id: userId },
+          data: {
+            aiProvider: aiProvider || DEFAULT_PROVIDER,
+            aiModel,
+            aiApiKey,
+          },
+        });
+
+        return { success: true };
+      } catch (error) {
+        return { serverError: "Failed to update settings" };
+      }
+    },
+  );
+
+export const updateDigestFrequencyAction = actionClient
+  .metadata({ name: "updateDigestFrequency" })
+  .schema(saveDigestFrequencyBody)
+  .action(
+    async ({ ctx: { emailAccountId }, parsedInput: { userFrequency } }) => {
+      try {
+        const emailAccount = await prisma.emailAccount.findUnique({
+          where: { id: emailAccountId },
+          include: { digestFrequency: true },
+        });
+
+        if (!emailAccount) {
+          return { serverError: "Email account not found" };
+        }
+
+        await prisma.$transaction(async (tx) => {
```

Comment:
i prefer not to use this form of transaction unless absolutely needed. prefer to use array instead:

```
$transaction([...])
```

1. Warning against them here: https://www.prisma.io/docs/orm/prisma-client/queries/transactions#interactive-transactions
2. Will force us to use Websockets if we want to use this: https://neon.com/docs/serverless/serverless-driver

In general we can almost always use non-interactive transactions so I'd default to that.